tac-operation-lifecycle/tac-operation-lifecycle-server/tests/helpers/mod.rs

```diff
@@ -0,0 +1,454 @@
+use std::sync::Arc;
+
+use blockscout_service_launcher::{test_database::TestDbGuard, test_server};
+use futures::StreamExt;
+use reqwest::Url;
+use tac_operation_lifecycle_logic::{client::settings::RpcSettings, database::TacDatabase};
+use tac_operation_lifecycle_server::Settings;
+
+pub async fn init_db(test_name: &str) -> TestDbGuard {
+    let db_name = format!("testdb_{test_name}");
+    TestDbGuard::new::<migration::Migrator>(db_name.as_str()).await
+}
+pub async fn init_tac_operation_lifecycle_server<F>(
+    db_url: String,
+    test_name: &str,
+    settings_setup: F,
+    realtime_boundary: u64,
+) -> Url
+where
+    F: Fn(Settings) -> Settings,
+{
+    let (settings, base) = {
+        let mut settings = Settings::default(db_url.clone());
+        let (server_settings, base) = test_server::get_test_server_settings();
+        settings.server = server_settings;
+        settings.metrics.enabled = false;
+        settings.tracing.enabled = false;
+        settings.jaeger.enabled = false;
+
+        (settings_setup(settings), base)
+    };
+
+    let test_db = init_db(test_name).await;
+    let db = Arc::new(TacDatabase::new(test_db.client()));
+
+    test_server::init_server(
+        move || {
+            tac_operation_lifecycle_server::run(settings, db.clone(), realtime_boundary.clone())
+        },
+        &base,
+    )
+    .await;
+    base
+}
+
+#[cfg(test)]
+mod tests {
+    use futures::stream::select_all;
+    use rand::Rng;
+    use std::{sync::Arc, time};
+
+    use super::*;
+    use migration::sea_orm::{ColumnTrait, EntityTrait, QueryFilter};
+    use sea_orm::Database;
+    use tac_operation_lifecycle_entity::interval;
+    use tac_operation_lifecycle_logic::{
+        client::Client, database::OrderDirection, settings::IndexerSettings, Indexer, IndexerJob,
+    };
+    use tokio::sync::Mutex;
+    use tracing::Instrument;
+
+    #[tokio::test]
+    async fn test_save_intervals() {
+        let db = init_db("save_intervals").await;
+        let conn_with_db = Database::connect(&db.db_url()).await.unwrap();
+
+        let catchup_interval = time::Duration::from_secs(rand::thread_rng().gen_range(1..100));
+        let tasks_number = rand::thread_rng().gen_range(1..100);
+        let lag = tasks_number * catchup_interval.as_secs();
+        let current_epoch = time::SystemTime::now()
+            .duration_since(time::UNIX_EPOCH)
+            .unwrap()
+            .as_secs();
+        let start_timestamp = current_epoch - lag;
+        tracing::debug!("start_timestamp: {}", start_timestamp);
```

Comment:
do these logs display properly? I don't see tracing being initialized before that.
maybe it's initialized by other tests when ran together, but in that case the logs won't work if this test is ran alone.
`let _ = tracing_subscriber::fmt::try_init();`